src/libawkward/array/UnionArray.cpp

```diff
@@ -531,12 +532,38 @@ namespace awkward {
 
   template <typename T, typename I>
   const Index64 UnionArrayOf<T, I>::count64() const {
-    throw std::runtime_error("FIXME: UnionArray::count64");
+    int64_t len = contents_.size();
+    Index64 tocount(len);
+    int64_t indx(0);
+    for (auto content : contents_) {
+      Index64 toappend = content.get()->count64();
+      tocount.ptr().get()[indx++] = toappend.length();
+    }
+    return tocount;
   }
 
   template <typename T, typename I>
   const std::shared_ptr<Content> UnionArrayOf<T, I>::count(int64_t axis) const {
-    throw std::runtime_error("FIXME: UnionArray::count");
+    if (axis == 0) {
+#ifdef _MSC_VER
+      std::string format = "q";
+#else
+      std::string format = "l";
+#endif
+
+      Index64 tocount = count64();
+      std::vector<ssize_t> shape({ (ssize_t)tocount.length() });
+      std::vector<ssize_t> strides({ (ssize_t)sizeof(int64_t) });
+
+      return std::make_shared<NumpyArray>(Identities::none(), util::Parameters(), tocount.ptr(), shape, strides, 0, sizeof(int64_t), format);
+    }
+    else {
+      std::vector<std::shared_ptr<Content>> contents;
+      for (auto content : contents_) {
+        contents.emplace_back(content.get()->count(axis));
+      }
+      return std::make_shared<UnionArrayOf<T, I>>(identities_, parameters_, tags_, index_, contents);
+    }
```

Comment:
@jpivarski - is it an to-do for this PR?

Reply:
If `UnionArray::simplify` was not done, I would say"no," but I finished it yesterday, so if you can easily merge to master, you can get it and put a call to it here.

On the last line, instead of constructing a `std::shared_ptr<UnionArray>`, construct a `UnionArray` in the stack and return its `unionarray.simplify()`. If the result has to be a union, it will return a union, but if any of the union's contents can be merged, including unions of unions, it will do so.

This will be standard procedure when returning unions: instead of avoiding creating unnecessarily complex structures, we write something general and then call `simplify`. `IndexedArrays` have a `simplify`, too, which prevents `IndexedArrays` of `IndexedArrays`.

Reply:
done